packages/hardhat-core/src/internal/hardhat-network/provider/context/rethnet.ts

```diff
@@ -133,11 +136,18 @@ export class RethnetEthContext implements EthContextAdapter {
     const limitContractCodeSize =
       config.allowUnlimitedContractSize === true ? 2n ** 64n - 1n : undefined;
 
+    const vmStub = {
```

Comment:
This has to be injected to both the Rethnet VM adapter and the Rethnet mempool, as some `solidty-coverage` functionality mines blocks.